inc/saitam.h

```diff
@@ -0,0 +1,890 @@
+ */
+typedef enum _sai_tam_stat_attr_t
+{
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_STAT_ATTR_START,
+
+    /**
+     * @brief Monitored object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_TAM_STAT_ATTR_PARENT_ID,
+
+    /**
+     * @brief Monitored object type
+     * Eg.: SAI_OBJECT_TYPE_BUFFER_POOL, SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP
+     * Not used for input. Optional for output.
+     *
+     * @type sai_object_type_t
+     * @flags READ_ONLY
+     */
+    SAI_TAM_STAT_ATTR_PARENT_OBJECT_TYPE,
+
+    /**
+     * @brief Counter
+     * Eg.: SAI_INGRESS_PRIORITY_GROUP_STAT_CURR_OCCUPANCY_BYTES
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_TAM_STAT_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_TAM_STAT_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TAM_STAT_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TAM_STAT_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_stat_attr_t;
+
+/**
+ * @brief Create and return a TAM stat id object
+ *
+ * This creates a TAM stat id object.
+ *
+ * @param[out] tam_id TAM stat id object
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count number of attributes
+ * @param[in] attr_list array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_stat_create_fn)(
+        _Out_ sai_object_id_t *stat_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ *
+ * @brief Deletes a specified tam stat id object.
+ *
+ * @param[in] stat_id TAM object to be removed.
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_stat_remove_fn)(
+        _In_ sai_object_id_t *stat_id);
+
+/**
+ * @brief Set TAM stat id object attribute value(s).
+ *
+ * @param[in] stat_id TAM stat id
+ * @param[in] attr attribute to set
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_stat_attribute_set_fn)(
+        _In_ sai_object_id_t stat_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get values for specified TAM stat id attributes.
+ *
+ * @param[in] stat_id - tam stat id object id
+ * @param[in] attr_count - number of attributes
+ * @param[inout] attr_list - array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_stat_attribute_get_fn)(
+        _In_ sai_object_id_t stat_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief TAM Statistic
+ *
+ * Identifies a specific counter within the SAI object hierarchy
+ * and provides the current value of the counter
+ */
+typedef struct _sai_tam_statistic_t
+{
```

Comment:
This could be attribute of  tam_stat object then thisnstructure istnieje not needed, And for grtting multple stara YouTube can add bulk API